hedera-node/test-clients/src/main/java/com/hedera/services/yahcli/commands/contract/evminfo/CodeLine.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.yahcli.commands.contract.evminfo;
+
+import com.hedera.services.yahcli.commands.contract.evminfo.Assembly.Columns;
+import com.hedera.services.yahcli.commands.contract.evminfo.Assembly.Variant;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Arrays;
+
+/**
+ * Represents an instruction line in the generated assembly
+ *
+ * <p>Instruction lines have an opcode, an optional operand (an array of bytes), an offset of where
+ * the instruction is in the bytecode, and an optional comment.
+ */
+public record CodeLine(int codeOffset, int opcode, byte[] operandBytes, String eolComment) implements Assembly.Code {
+
+    public CodeLine {
+        opcode = opcode & 0xFF; // convert to unsigned (in case it's not)
+        // De-null optional arguments
+        operandBytes = null != operandBytes ? operandBytes : new byte[0];
+        eolComment = null != eolComment ? eolComment : "";
+    }
+
+    @Override
+    @NonNull
+    public String mnemonic() {
+        return Opcodes.getOpcode(opcode).mnemonic();
+    }
+
+    @Override
+    public void formatLine(@NonNull StringBuilder sb) {
+        if (Variant.DISPLAY_CODE_OFFSET.getValue()) {
+            extendWithBlanksTo(sb, Columns.CODE_OFFSET.getColumn());
+            sb.append("%5X".formatted(codeOffset));
+        }
+
+        if (Variant.DISPLAY_OPCODE_HEX.getValue()) {
+            extendWithBlanksTo(sb, Columns.OPCODE.getColumn());
+            sb.append("  %02X".formatted(opcode()));
```

Comment:
Unknown spaces throw me off. Why this number of characters and why for this variant?